contracts/NFTVesting/LinearVestingNFTTimeLock.sol

```diff
@@ -0,0 +1,122 @@
+    uint256 private immutable _maxDiscount;
+
+    // Duration that token will vest
+    uint256 private _maxDuration;
+
+
+    /**
+     * @dev Deploys a timelock instance that is able to hold the token specified, and will only release it to
+     * `beneficiary_` when {release} is invoked after `releaseTime_`. The release time is specified as a Unix timestamp
+     * (in seconds).
+     */
+    constructor(
+        IERC721 nft_,
+        uint256 tokenId_,
+        address beneficiary_,
+        uint256 releaseTime_,
+        uint256 maxDiscount_,
+        uint256 maxDuration_
+    ) {
+        require(releaseTime_ > block.timestamp, "BasicNFTTimelock: release time is before current time");
+        _nft = nft_;
+        _tokenId = tokenId_;
+        _beneficiary = beneficiary_;
+        _releaseTime = releaseTime_;
+        _maxDiscount = maxDiscount_;
+        _maxDuration = maxDuration_;
+    }
+
+    /**
+     * @dev Returns the smart contract NFT.
+     */
+    function nft() public view virtual returns (IERC721) {
+        return _nft;
+    }
+
+    /**
+     * @dev Returns the token ID of the NFT being held.
+     */
+    function tokenId() public view virtual returns (uint256) {
+        return _tokenId;
+    }
+
+    /**
+     * @dev Returns the beneficiary that will receive the NFT.
+     */
+    function beneficiary() public view virtual returns (address) {
+        return _beneficiary;
+    }
+
+    /**
+     * @dev Returns the time when the NFT are released in seconds since Unix epoch (i.e. Unix timestamp).
+     */
+    function releaseTime() public view virtual returns (uint256) {
+        return _releaseTime;
+    }
+
+     /**
+     * @dev Returns discount percentage for achieved from vesting
+     */
+
+    function vestedDiscountPercentage() public view returns (uint256) {
+        if (block.timestamp < _releaseTime) {
+            return 0;
+        }
+        return _maxDiscount * (block.timestamp - _releaseTime) / _maxDuration;
+    }
+
+    /**
+     * @dev Returns discount for achieved from vesting
+     */
+    function vestedDiscount() public view returns (uint256) {
+        uint256 currentBalance = address(this).balance;
+        uint256 discount = currentBalance * vestedDiscountPercentage();
+        
+        return discount;
+    }
+
+    /**
+     * @dev Transfers NFT held by the timelock to the beneficiary. Will only succeed if invoked after the release
+     * time. Sends the discount in Eth to the beneficiary.
+     */
+    function release() public virtual {
+        require(block.timestamp >= releaseTime(), "TimeLock: current time is before release time");
+        require(nft().ownerOf(tokenId()) == address(this), "TimeLock: no NFT to release for user");
+        
+        uint256 ethDiscount = vestedDiscount();
+        (bool sent, ) = beneficiary().call{value: ethDiscount}(""); 
+        require(sent, "Failed to send Ether");
+
+        nft().safeTransferFrom(address(this), beneficiary(), tokenId());
```

Comment:
Same re: ability to make this atomic?

Reply:
not too sure how i can make this function atomic?

is it the call and require function?